youtube_dl/extractor/funimation.py

```diff
@@ -227,6 +249,81 @@ def _real_extract(self, url):
+            'title': 'A Certain Scientific Railgun'
+        },
+        'playlist_count': 48
+    }, {
+        'url': 'http://www.funimation.com/shows/hacksign/home',
+        'info_dict': {
+            'id': 'hacksign',
+            'description': '	Tsukasa wakes up inside The World, a massive online role-playing game full of magic and monsters, and finds himself unable to log out. With no knowledge of what’s happening in the real world, Tsukasa must discover how he ended up stuck in the game, and what connection he has with the fabled Key of the Twilight—an item that’s rumored to grant ultimate control over the digital realm.',
+            'title': '.hack//SIGN'
+        },
+        'playlist_count': 56
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+
+        user_agent = self._extract_cloudflare_session_ua(url)
+
+        # Use series page to get ID number and title / description
+        series_url = self._search_regex(self._VALID_URL, url, 'series URL')
+        request = sanitized_Request(series_url)
+        request.add_header('User-Agent', user_agent)
+        webpage = self._download_webpage(request, display_id, 'Downloading series webpage')
+
+        # Parseable show data stored as a JavaScript variable
+        playlist = self._parse_json(
+            self._search_regex(
+                r'var\s+playersData\s*=\s*(\[.+?\]);\n',
+                webpage, 'players data'),
+            display_id)[0]['playlist'][0]
+
+        def pagefunc(pagenum):
+            # Internal Funimation endpoint for getting paginated video list HTML
+            request = sanitized_Request(
+                'https://www.funimation.com/shows/viewAllFiltered?section=episodes&showid={}&offset={}'
+                .format(playlist.get('showId'), pagenum * 20))
+            request.add_header('User-Agent', user_agent)
+            episode_list = self._download_json(
+                request, display_id, 'Downloading episode list from {}'.format(pagenum * 20))['main']
+
+            # There are multiple instances of each video URL, so filter for unique URLs
+            # while keeping the order of the episodes
+            urls_seen = set()
+            episode_paths = re.finditer(
+                r'(?s)<a href="(' + FunimationIE._VALID_URL + r')"',
+                episode_list)
+            episode_paths = [
+                path.group(1) for path in episode_paths
+                if not (path.group(1) in urls_seen or urls_seen.add(path.group(1)))]
+
+            return [InfoExtractor.url_result(ep, 'Funimation') for ep in episode_paths]
```

Comment:
`self.url_result` is enough. Also `FunimationIE.ie_key()` instead of hardcoded string.